Rules/UseUsingScopeModifierInNewRunspaces.cs

```diff
@@ -0,0 +1,493 @@
+                var scriptBlockParameterAst = commandAst.CommandElements[
+                        commandAst.CommandElements.IndexOf(scriptBlockExpressionAst) - 1] as CommandParameterAst;
+
+                if (IsInlineScriptBlock(cmdName) || 
+                    IsJobScriptBlock(cmdName, scriptBlockParameterAst) ||
+                    IsForeachScriptBlock(cmdName, scriptBlockParameterAst) ||
+                    IsInvokeCommandComputerScriptBlock(cmdName, commandAst) ||
+                    IsDSCScriptResource(cmdName, commandAst))
+                {
+                    AnalyzeScriptBlock(scriptBlockExpressionAst);
+                    return AstVisitAction.SkipChildren;
+                }
+
+                if (IsInvokeCommandSessionScriptBlock(cmdName, commandAst))
+                {
+                    if (!TryGetSessionNameFromInvokeCommand(commandAst, out var sessionName))
+                    {
+                        return AstVisitAction.Continue;
+                    }
+
+                    IEnumerable<VariableExpressionAst> varsInLocalAssignments = FindVarsInAssignmentAsts(scriptBlockExpressionAst);
+                    if (varsInLocalAssignments != null)
+                    {
+                        AddAssignedVarsToSession(sessionName, varsInLocalAssignments);
+                    }
+
+                    GenerateDiagnosticRecords(
+                        FindNonAssignedNonUsingVarAsts(
+                            scriptBlockExpressionAst, 
+                            GetAssignedVarsInSession(sessionName)));
+
+                    return AstVisitAction.SkipChildren;
+                }
+                
+                return AstVisitAction.Continue;
+            }
+
+            /// <summary>
+            /// FindVarsInAssignmentAsts: Retrieves all assigned variables from an Ast:
+            /// Example: `$foo = "foo"` ==> the VariableExpressionAst for $foo is returned
+            /// </summary>
+            /// <param name="ast"></param>
+            private static IEnumerable<VariableExpressionAst> FindVarsInAssignmentAsts(Ast ast)
+            {
+                // Find all variables that are assigned within this ast
+                foreach (Ast foundAst in ast.FindAll(VarsInAssignments, true))
+                {
+                    var assignment = foundAst as AssignmentStatementAst;
+                    
+                    if (assignment.Left is VariableExpressionAst variable)
+                    {
+                        yield return variable;
+                    }
+                    else if (assignment.Left is ConvertExpressionAst conversion)
+                    {
+                        yield return conversion.Child as VariableExpressionAst;
+                    }
+                };
+            }
+
+            /// <summary>
+            /// VarsInAssignments: helper function to prevent allocation of closures for FindAll predicate.

```

Comment:
I know it's the convention in the rule comments, but there's no need to repeat the method name in the comment. The comment should add information that isn't present in the name or type signature of the thing it's attached to.

Reply:
No need to change it